File: backend/src/main/java/br/com/fleetmanager/model/OrderOfServiceSST.java
package br.com.fleetmanager.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "order_of_service_sst")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderOfServiceSST {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, length = 2000)
    private String description;

    @Column(nullable = false)
    private String status;

    @Column(nullable = false)
    private String responsible;

    @Column(name = "issue_date", nullable = false)
    private LocalDateTime issueDate;

    @Column(name = "execution_date")
    private LocalDateTime executionDate;

    @Column(name = "document_url")
    private String documentUrl;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
} 

File: frontend/src/types/orderOfServiceSST.ts
export interface OrderOfServiceSST {
  id: string;
  title: string;
  description: string;
  status: string;
  responsible: string;
  issueDate: string;
  executionDate?: string;
  documentUrl?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrderOfServiceSSTDTO {
  title: string;
  description: string;
  status: string;
  responsible: string;
  issueDate: string;
  executionDate?: string;
  documentUrl?: string;
}

export interface UpdateOrderOfServiceSSTDTO extends Partial<CreateOrderOfServiceSSTDTO> {
  id: string;
}

export interface OrderOfServiceSSTFilters {
  status?: string;
  responsible?: string;
  issueDateFrom?: string;
  issueDateTo?: string;
} 